### DISCUSSION TITLE
Star rating balancing for Easy/Normal difficulties

### DISCUSSION THREAD

**Discussion by Serizawa-Haruki** [ppy/osu#24483] [Q&A]
While star rating has never been a perfectly accurate measurement for difficulty, I have noticed that for low difficulties in osu! standard (Easy/Normal in particular) it tends to be especially misrepresentative. My main concern is that the average SR difference between Easy and Normal is very low compared to the difference between other diffs like Normal-Hard or Hard-Insane. For example, the last 10 ranked mapsets have the following differences: 0.61, 0.47, 0.72, 0.47, 0.43, 0.62, 0.25, 0,59, 0.66, 0.56 - an average of 0.538. On the same mapsets, the average difference on the 2 higher difficulty gaps are 1.056 and 0.996 respectively - almost twice as much. Of course this is just a very small dataset but I'm only using it as a quick reference. And obviously all of this depends on a number of factors like how each difficulty is mapped, the song's BPM, etc. and I suppose nowadays Easy diffs are typically not that much easier than Normals due to heavy 1/1 rhythm usage, however it seems like the star rating range for Easies is relatively low, in most cases between 1.5 and 2.0.
I looked at my own maps and if I compare those with similar BPM, the same trend can be seen. For example, https://osu.ppy.sh/beatmapsets/932291#osu/2166563 and https://osu.ppy.sh/beatmapsets/1279201#osu/2707636 are 1.75 and 1.73 respectively (Easy). Meanwhile https://osu.ppy.sh/beatmapsets/829145#osu/1788569 and https://osu.ppy.sh/beatmapsets/1487122#osu/3073331 are 2.27 and 2.35 (Normal). Despite the significant leap in rhythm density, the star rating is again only around 0.65 higher. Even extreme cases like https://osu.ppy.sh/beatmapsets/1797691#osu/3692534 vs. https://osu.ppy.sh/beatmapsets/1680649#osu/3920600 where the Easy is very simple while the Normal is much harder and more complex (and the map is longer) only show a difference of 0.78 which is significantly less than the average difficulty gap (around 1.0 or even more).
I think the reason for this could be that spacing plays a major role in SR calculation, but the main factor that distinguishes Easies from Normals is the rhythm and not so much the distance between the objects. After all, you can't reduce the spacing too much in an Easy because then 1/1 gaps would overlap which makes reading harder. The added 1/2 rhythms found in Normal diffs might be underweighted because they are mostly expressed in the form of sliders and passive rhythms which are considered less difficult than active rhythms by the algorithm as far as I know. Active 1/2 rhythms are also often stacked which again scores low on the spacing factor. I think these aspects should be taken into account for the calculation of star rating for low diffs, meaning that not only spacing/aim/click speed are considered but also rhythm density and complexity. That way Easy difficulties with simple rhythm would get a slightly lower rating while Normal difficulties with high rhythm density would get a slightly higher one, this would help bring the gap in line with the higher difficulty gaps.

**Comment by Rekunan** [comment]
Coincidentally, the changes u propose are already planned, as some issues are present in all difficulties (such as rhythm complexity) and are just taking a long time to get the values perfect.

**Comment by Givikap120** [comment]
star rating is extremely inaccurate for low difficulties
and considering that this is not a priority - this will not be fixed any time soon

**Reply by Rekunan** [reply]
I think it's dishonest to just leave it at that. Like how I said in my answer, there are already plans to implement some of the proposed changes more generally, and I believe it'll trickle down to improving low difficulties.

**Reply by Serizawa-Haruki** [reply]
This is exactly the problem I think, star rating and pp reworks are focused exclusively on higher difficulties despite a large part of the playerbase being in the lower level range. Just because there is less interest in these difficulties due to having pp values that are usually not particularly impressive, doesn't mean that they don't need/deserve proper balancing as well. While I understand it's not a priority, it should still be addressed at some point and so far I haven't really seen any changes regarding this.
